src/plugins/auto_batch/src/async_infer_request.cpp

```diff
@@ -9,57 +9,57 @@
 namespace ov {
 namespace autobatch_plugin {
 
-AsyncInferRequest::AsyncInferRequest(const SyncInferRequest::Ptr& inferRequest,
-                                     InferenceEngine::SoIInferRequestInternal& inferRequestWithoutBatch,
-                                     const InferenceEngine::ITaskExecutor::Ptr& callbackExecutor)
-    : AsyncInferRequestThreadSafeDefault(inferRequest, nullptr, callbackExecutor),
-      m_infer_request_without_batch(inferRequestWithoutBatch),
-      m_sync_infer_request{inferRequest} {
+AsyncInferRequest::AsyncInferRequest(const std::shared_ptr<SyncInferRequest>& request,
+                                     std::shared_ptr<ov::IAsyncInferRequest> request_without_batch,
+                                     const std::shared_ptr<ov::threading::ITaskExecutor>& callback_executor)
+    : ov::IAsyncInferRequest(request, nullptr, callback_executor),
+      m_sync_request(request),
+      m_request_without_batch(request_without_batch) {
     // this executor starts the inference while  the task (checking the result) is passed to the next stage
-    struct ThisRequestExecutor : public InferenceEngine::ITaskExecutor {
+    struct ThisRequestExecutor : public ov::threading::ITaskExecutor {
         explicit ThisRequestExecutor(AsyncInferRequest* _this_) : _this{_this_} {}
-        void run(InferenceEngine::Task task) override {
-            auto& workerInferRequest = _this->m_sync_infer_request->m_batched_request_wrapper;
-            std::pair<AsyncInferRequest*, InferenceEngine::Task> t;
+        void run(ov::threading::Task task) override {
+            auto workerInferRequest = _this->m_sync_request->m_batched_request_wrapper;
+            std::pair<AsyncInferRequest*, ov::threading::Task> t;
             t.first = _this;
             t.second = std::move(task);
-            workerInferRequest._tasks.push(t);
+            workerInferRequest->_tasks.push(t);
             // it is ok to call size() here as the queue only grows (and the bulk removal happens under the mutex)
-            const int sz = static_cast<int>(workerInferRequest._tasks.size());
-            if (sz == workerInferRequest._batchSize) {
-                workerInferRequest._cond.notify_one();
+            const int sz = static_cast<int>(workerInferRequest->_tasks.size());
+            if (sz == workerInferRequest->_batch_size) {
+                workerInferRequest->_cond.notify_one();
             }
         };
         AsyncInferRequest* _this = nullptr;
```

Comment:
is it possible to avoid raw pointers at all?

Reply:
@vurusovs  Any suggestions? I just follow the original way of api 1.0.

Reply:
Something like next:
```
    struct ThisRequestExecutor : public ov::threading::ITaskExecutor {
        explicit ThisRequestExecutor(std::shared_ptr<AsyncInferRequest> _this_) : _this{_this_} {}
        void run(ov::threading::Task task) override {
            auto workerInferRequest = _this->m_sync_request->m_batched_request_wrapper;
            std::pair<std::shared_ptr<AsyncInferRequest>, ov::threading::Task> t;
            t.first = _this;
            t.second = std::move(task);
            workerInferRequest->_tasks.push(t);
            // it is ok to call size() here as the queue only grows (and the bulk removal happens under the mutex)
            const int sz = static_cast<int>(workerInferRequest->_tasks.size());
            if (sz == workerInferRequest->_batch_size) {
                workerInferRequest->_cond.notify_one();
            }
        };
        std::shared_ptr<AsyncInferRequest> _this = nullptr;
    };
    m_pipeline = {{/*TaskExecutor*/ std::make_shared<ThisRequestExecutor>(shared_from_this()), /*task*/ [this] {
```
It's a little bit safer in terms of RAII

Reply:
Try this change. Here shared_from_this is called without having std::shared_ptr owning the caller, which caused throw "bad_weak_ptr" exception.